### PR TITLE
docs: clip benchmark on zeroshot classification and retrieval tasks

### PR DESCRIPTION
**Pull request by ZiniuYu**

### CODE REVIEW

File: docs/user-guides/benchmark.rst [hunk 1]
```diff
@@ -0,0 +1,294 @@
...
+In order to evaluate the performance of different CLIP models, we conducted a benchmark on a series of tasks using different datasets. 
+We fork the `CLIP benchmark repo <https://github.com/LAION-AI/CLIP_benchmark>`_, slightly modify the codebase and apply it to all Vision Transformers (ViT) and ResNet (RN) CLIP models. 
+You can find the benchmark results in the following tables. 
+The best results are highlighted in bold. 
+They can be used as a guide to choose the best model for your application.
+
+Basic statistics
+----------------
+
+In the table below, we include the disk usage (in delta) and the peak RAM and VRAM usage (in delta) when running on a single Nvidia TITAN RTX GPU (24GB VRAM) for a series of text and image encoding tasks with ``batch_size=8`` using PyTorch runtime.
```
Comment (jemmyshin) [inline docs/user-guides/benchmark.rst:13]: please also specify the cpu/memory info, also the version of `jina`, `cas`, `docarray`...

File: docs/user-guides/benchmark.rst [hunk 2]
```diff
@@ -0,0 +1,294 @@
...
+| ViT-g-14::laion2b_s12b_b42k      | 0.696      | **0.811** | **0.851**  | 0.839     | **0.682** | 0.776      | 0.943     | **0.962** | **0.603** | 0.648     | 0.718     | 0.560          | 0.580       | **0.332**   | 0.175          | 0.036             | 0.031                | 0.060             | 0.115               | 0.190     | 0.138          |
++----------------------------------+------------+-----------+------------+-----------+-----------+------------+-----------+-----------+-----------+-----------+-----------+----------------+-------------+-------------+----------------+-------------------+----------------------+-------------------+---------------------+-----------+----------------+
+
+From the table, we observe that the ViT models still outperform the RN models in most tasks, except for the Patch Camelyon dataset where ``RN50::openai`` has the best top-1 accuracy of 0.636, and the KITTI/distance dataset where ``RN50::yfcc15m`` has the best result of 0.336.
```
Comment (jemmyshin) [inline docs/user-guides/benchmark.rst:216]: There are some cases where RN models beat ViT, such as `Patch Camelyon`, `KITTI/distance`. Is there any reason that they perform like this way?

File: docs/user-guides/benchmark.rst [hunk 3]
```diff
@@ -0,0 +1,294 @@
...
+Similar to retrieval results, the ``ViT-H-14::laion2b_s32b_b79k`` model and ``ViT-g-14::laion2b_s12b_b42k`` model still have the best or close to the best results on 12/21 zero-shot classification tasks.
+All models tend to perform well on ImageNetV2, VOC2007, VTAB natural and VTAB specialized (except for Retinopathy) datasets, whereas they perform poorly on VTAB structured datasets.
+We do not observe any significant difference between the ViT models of the same base model.
```
Comment (jemmyshin) [inline docs/user-guides/benchmark.rst:219]: what does this mean?

Comment (ZiniuYu) [reply docs/user-guides/benchmark.rst:219]: For ViT models in retrieval tasks, the results of the same base model are better on those pre-trained with larger datasets (e.g., ViT-B-32::openai vs ViT-B-32::laion400m_e31 vs ViT-B-32::laion2b-s34b-b79k).
This is not the same in classification tasks

**Review by numb3r3 — APPROVED** [review]

**Pull request merged**